### ISSUE TITLE
Cannot focus lozenges using keyboard

### ISSUE THREAD

**Issue by robertknight** [hypothesis/h#4079]
Related to https://github.com/hypothesis/h/issues/3991 , aside from supporting backspace to delete search terms, there are a couple of fundamental issues with keyboard accessibility of the search input:

1. The keyboard cannot be used to focus client-rendered lozenges
2. Although a server-rendered lozenge can be focused using the keyboard and removed by pressing the Enter key, there is no visual indication when the lozenge is focused
